backend/src/test/java/heartbeat/handler/AsyncMetricsDataHandlerTest.java

```diff
@@ -239,6 +245,69 @@ void shouldUpdateAllMetricDataWhenPreviousMetricsStatusIsNotNull() throws IOExce
 
 	}
 
+	@Nested
+	class UpdateAllMetricsCompletedInHandlerAtTheSameTime {
+
+		@RepeatedTest(100)
+		void shouldUpdateAllMetricDataAtTheSameTimeWhenPreviousMetricsStatusIsNotNull() throws IOException {
+			long currentTimeMillis = System.currentTimeMillis();
+			String currentTime = Long.toString(currentTimeMillis);
+			List<Integer> sleepTime = new ArrayList<>();
+			for (int i = 0; i < 3; i++) {
+				sleepTime.add(new Random().nextInt(100));
+			}
+			MetricsDataCompleted metricsDataCompleted = MetricsDataCompleted.builder()
+				.boardMetricsCompleted(false)
+				.doraMetricsCompleted(false)
+				.overallMetricCompleted(false)
+				.build();
+			asyncMetricsDataHandler.putMetricsDataCompleted(currentTime, metricsDataCompleted);
+
+			List<CompletableFuture<Void>> threadList = new ArrayList<>();
+
+			threadList.add(CompletableFuture.runAsync(() -> {
+				try {
+					TimeUnit.MILLISECONDS.sleep(sleepTime.get(0));
+				}
+				catch (InterruptedException e) {
+					throw new RuntimeException(e);
```

Comment:
Seems need to throw a certain type exception.

Reply:
has ignored the catch block.